draft-ietf-quic-transport.md

```diff
@@ -7486,6 +7483,28 @@ ECN to be disabled.  For those rare cases where marked packets are discarded by
 the path, the short duration of the testing period limits the number of losses
 incurred.
 
+# Sample Variable-Length Integer Decoding {#sample-varint}
+
+The pseudo-code in {{alg-varint}} shows how a variable-length integer can be
+read from a stream of bytes.  The function ReadVarint takes a single argument, a
+sequence of bytes to read from.
+
+~~~
+ReadVarint(data):
+  first_byte = data.peek()
+  prefix = first_byte >> 6
+  num_bytes = 1 << prefix
+  mask = (1 << (num_bytes * 8 - 2)) - 1
+
+  return data.read(num_bytes) & mask
```

Comment:
```suggestion
  v = data.next_byte()
  length = (1 << (v >> 6))
  v = v & 0x3f
  repeat length-1 times:
    v = (v << 8) + data.next_byte()
```

This is not efficient in any way (I implemented it this way and it is slow), but it should be clearer.  And it doesn't rely on peeking or a read() function that returns something other than a byte.

Reply:
I would just add comments above these two lines:
```
  // Length is encoded in the two most significant bits of the first byte;
  // see {{summary-of-integer-encoding}}.
  length = (1 << (v >> 6))
  // Remove the size encoding from the data
  v = v & 0x3f

Reply:
Note that moustaches in figures don't get expanded into section references, so you might want to lose the reference.